gemrb/core/Interface.cpp

```diff
@@ -612,15 +625,21 @@ void Interface::Main()
 	tick_t frame = 0;
 	tick_t time = GetMilliseconds();
 	tick_t timebase = time;
+	tick_t lastGameUpdate = 0;
+
 	double frames = 0.0;
 
 	do {
-		for (auto it = timers.begin(); it != timers.end();) {
-			if (it->IsRunning()) {
-				it->Update(time);
-				++it;
-			} else {
-				it = timers.erase(it);
+		bool doGameStateUpdate = time - lastGameUpdate >= 1000.0 / Time.ai_update_time;
+
+		if (doGameStateUpdate) {
+			for (auto it = timers.begin(); it != timers.end();) {
```

Comment:
I'm pretty sure these timers are realtime, so they don't need to be limited to game ticks.

Reply:
I'll better check this thoroughly.

Reply:
It's used by the GUI via SetTimer. Doesn't look like there'd be much difference either way.

Reply:
PyDoc best describes their use as "time interval in ticks". I can remove the check but I have no idea how this is meant to be understood.

Reply:
It would make a difference for some animation types (eg color anims), since you are limiting them to 30fps. Certainly it would make a difference to `ScrollView` since it uses `PointAnimation` to auto scroll.

Reply:
re `GemRB_SetTimer`, that documentation is out of date. these timers used to be ran by `GlobalTimer`.